pkg/workspace/project/project.go

```diff
@@ -40,8 +40,10 @@ type ProjectState struct {
 } // @name ProjectState
 
 type GitStatus struct {
-	CurrentBranch string        `json:"currentBranch" validate:"required"`
-	Files         []*FileStatus `json:"fileStatus" validate:"required"`
+	CurrentBranch   string        `json:"currentBranch" validate:"required"`
+	Files           []*FileStatus `json:"fileStatus" validate:"required"`
+	BranchPublished bool          `json:"branchPublished" validate:"optional"`
+	UnpushedCommits []CommitInfo  `json:"unpushedCommits" validate:"optional"`
```

Comment:
Instead of the UnpushedCommits being an array of objects, I suggest that we have `Ahead` and `Behind` numbers and that's it.